Marlin/temperature.cpp

```diff
@@ -583,6 +584,59 @@ float get_pid_output(int e) {
   }
 #endif
 
+#if MAX_TEMP_OVERSTOOT_TIME > 0
+  void check_heater_works(int8_t hh) {
+    enum state_t { FULL_ON_S, ON_S, OFF_S };
+    static state_t state[5] = { ON_S, ON_S, ON_S, ON_S, ON_S, };
+    static millis_t timerc[5] = {0, 0, 0, 0, 0};
+    static float initial_temp[5] = {0.0, 0.0, 0.0, 0.0, 0.0};
+
+    uint8_t h = hh + 1;
+//    SERIAL_PROTOCOLPGM("Nr:"); SERIAL_PROTOCOL((int)h); SERIAL_PROTOCOLPGM("Timer:"); SERIAL_PROTOCOL((timerc[h]-millis())); SERIAL_PROTOCOLPGM(" State:"); SERIAL_PROTOCOL(state[h]); SERIAL_PROTOCOLPGM(" PWM:"); SERIAL_PROTOCOL((int)((h) ? soft_pwm[hh] : soft_pwm_bed)); SERIAL_PROTOCOLPGM(" I temp:"); SERIAL_PROTOCOL(initial_temp[h]); SERIAL_PROTOCOLPGM(" temp:"); SERIAL_PROTOCOLLN(((h) ? current_temperature[hh] : current_temperature_bed));
+    if (((h) ? soft_pwm[hh] : soft_pwm_bed) == ((h) ? PID_MAX >> 1: MAX_BED_POWER >> 1))
+      if (state[h] != FULL_ON_S) {
+        state[h]  = FULL_ON_S;
+        timerc[h] = millis() + MAX_TEMP_OVERSTOOT_TIME * 1000UL;
+        initial_temp[h] = ((h) ? current_temperature[hh] : current_temperature_bed) - MAX_SENSOR_NOISE_C;
+        return;
+      }
+      else {
+        if (millis() > timerc[h])
+          if(((h) ? current_temperature[hh] : current_temperature_bed) <= initial_temp[h]) {
+              _temp_error(hh, PSTR(MSG_HEATING_FAILED), PSTR(MSG_HEATING_FAILED_LCD));
```

Comment:
The _temp_error messages should end with "Heater ID: " because the number or "Bed" is appended. `MSG_HEATING_FAILED` is missing this suffix. (Or, probably there should be a "Heater ID: " message that is also printed immediately after, then these errors don't all need to include it, which wastes bytes.